s29986-PrzemyslawKosiedowski/zajecia3/zadanie_domowe.txt

```diff
@@ -0,0 +1,100 @@
+Do wykonania zadań skorzystałem z dokumentacji basha:
+https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html
+oraz mana.
+
+===============================================================================
+Zadanie domowe
+===============================================================================
+W przypadku bardzo podobnych rozwiazań nie będę ich uznawał.
+
+Przygotuj wyrażenie regularne, które będzie sprawdzało, czy tekst jest
+poprawnie napisanym polskim imieniem (od wielkiej litery, może mieć tylko
+litery i zawierać polskie znaki).
+
+Przygotuj wyrażenie regularne sprawdzające czy tekst jest kodem pocztowym
+(cały tekst, czyli zaczyna się od cyfry i kończy się cyfrą).
+
+Przygotuj wyrażenie regularne sprawdzające e-mail.
+
+
+Napisz skrypt który będzie pobierał jeden argument z linii komend i wypisywał
+informację o tym, czy jest to imie, kod pocztowy, czy też email. Na przykład
+
+./coto Janusz
+
+imię
+
+./coto 68-221
+
+kod pocztowy
+
+./coto 'muniek+spam@gmail.com'
+
+email
+-------------------------------------------------------------------------------
+
+Wyrażenie 1:
+^([A-Z]|[ĄĆĘŁŃÓŚŹŻ])([a-z]|[ąćęłńóśźż]){1,}$
+
+gdzie:
+^ i $ - początek i koniec wyrażenia
+([A-Z]|[ĄĆĘŁŃÓŚŹŻ]) - dopasowuje jedną wielką literę, uwzględnia polskie znaki
+([a-z]|[ąćęłńóśźż]) - dopasowuje jedną lub więcej liter, uwzględnia polskie
+	znaki
+
+
+Wyrażenie 2:
+^[0-9]{2}[-]{1}[0-9]{3}$
+
+gdzie
+^[0-9]{2} - dopasowuje dwie cyfry na początku
+[-]{1} - po czym dopasowuje jeden myślnik
+[0-9]{3}$ - na sam koniec dopasowuje trzy cyfry
+
+
+Wyrażenie 3:
+^[a-zA-Z0-9._%+-]{1,}@[a-zA-Z0-9.-]{1,}\.[a-zA-Z]{2,}$ 
```

Comment:
Zapis `{1,}` jest tożsamy z innym znacznie prostszym.
Tak samo jak zapis `{0,}` jest także tożsamy z innym prostszym i bardzo często używanym. Z uwagi na to, że jest tak bardzo popularny skrócony zapis lepiej przyswoić sobie te 2 powyższe wymienione przeze mnie przypadki formy skróconej (do poszukania w dokumentacji o czym mowa).